src/transaction/utils.ts

```diff
@@ -63,25 +64,35 @@ function prepareTransaction(txJSON: any, api: RippleAPI,
     const multiplier = instructions.signersCount === undefined ? 1 :
       instructions.signersCount + 1
     if (instructions.fee !== undefined) {
+      const fee = new BigNumber(instructions.fee)
+      if (fee.greaterThan('2') && instructions.allowHighFee !== true) {
+        const errorMessage = `Fee of ${fee.toString(10)} XRP exceeds ` +
+          'soft limit of 2 XRP. To use this fee, set ' +
+          '`instructions.allowHighFee = true`.'
+        throw new ValidationError(errorMessage)
+      }
       txJSON.Fee = scaleValue(common.xrpToDrops(instructions.fee), multiplier)
       return Promise.resolve(txJSON)
     }
     const cushion = api._feeCushion
-    return api.getFee(cushion).then(fee => {
+    return api.getFee(cushion, null).then(fee => {
```

Comment:
Do we need to add the `maximum` argument to `getFee` if we're not using it?

Reply:
We are using it. If the argument is undefined, it will default to `1`, but with `null`, no maximum is imposed.

Reply:
That should have said: Do we need `getFee` to enforce a maximum (which is configurable via the `maximum` argument`) if we're going to bypass it?

Reply:
I think so. This is not the only place `getFee` is used; it's also a public API.